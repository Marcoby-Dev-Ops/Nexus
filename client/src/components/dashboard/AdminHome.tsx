import React from 'react';

// Mock data
const users = [
  { id: 1, name: 'Alice', email: 'alice@email.com', role: 'Admin', status: 'Active' },
  { id: 2, name: 'Bob', email: 'bob@email.com', role: 'Manager', status: 'Active' },
  { id: 3, name: 'Carol', email: 'carol@email.com', role: 'User', status: 'Invited' },
];
const teams = [
  { id: 1, name: 'Sales', members: 5 },
  { id: 2, name: 'Finance', members: 3 },
  { id: 3, name: 'Operations', members: 4 },
];
const activityLog = [
  { action: 'Invited user', who: 'Alice', date: '2024-06-10' },
  { action: 'Changed logo', who: 'Bob', date: '2024-06-09' },
];
const aiInsights = [
  'Carol has not accepted her invite (3 days).',
  'No security issues detected.',
];

export default function AdminHome() {
  return (
    <div className="p-8 space-y-8">
      {/* Quick Actions */}
      <QuickActions />
      {/* User Management */}
      <div className="rounded-xl border p-4 bg-card">
        <h3 className="font-semibold mb-2">User Management</h3>
        <table className="min-w-full text-sm">
          <thead>
            <tr>
              <th className="text-left p-2">Name</th>
              <th className="text-left p-2">Email</th>
              <th className="text-left p-2">Role</th>
              <th className="text-left p-2">Status</th>
            </tr>
          </thead>
          <tbody>
            {users.map((user) => (
              <tr key={user.id} className="border-t">
                <td className="p-2">{user.name}</td>
                <td className="p-2">{user.email}</td>
                <td className="p-2">{user.role}</td>
                <td className="p-2">{user.status}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
      {/* Team Management */}
      <div className="rounded-xl border p-4 bg-card">
        <h3 className="font-semibold mb-2">Team Management</h3>
        <ul className="divide-y">
          {teams.map((team) => (
            <li key={team.id} className="py-2 flex justify-between">
              <span>{team.name}</span>
              <span className="text-xs text-muted-foreground dark:text-gray-200">{team.members} members</span>
            </li>
          ))}
        </ul>
      </div>
      {/* System Settings */}
      <div className="rounded-xl border p-4 bg-card">
        <h3 className="font-semibold mb-2">System Settings</h3>
        <ul className="list-disc pl-5">
          <li>Branding: <span className="text-muted-foreground dark:text-gray-200">Nexus OS</span></li>
          <li>Integrations: <span className="text-muted-foreground dark:text-gray-200">QuickBooks, Slack</span></li>
          <li>Preferences: <span className="text-muted-foreground dark:text-gray-200">Dark mode enabled</span></li>
        </ul>
      </div>
      {/* Activity Log & AI Insights */}
      <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        <div className="rounded-xl border p-4 bg-card">
          <h3 className="font-semibold mb-2">Activity Log</h3>
          <ul className="divide-y">
            {activityLog.map((a, i) => (
              <li key={i} className="py-2">
                {a.action} by <span className="font-medium">{a.who}</span> <span className="text-xs text-muted-foreground dark:text-gray-200">({a.date})</span>
              </li>
            ))}
          </ul>
        </div>
        <div className="rounded-xl border p-4 bg-card">
          <h3 className="font-semibold mb-2">AI Insights</h3>
          <ul className="list-disc pl-5">
            {aiInsights.map((insight, i) => (
              <li key={i}>{insight}</li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
}

// Stub: QuickActions
function QuickActions() {
  return (
    <div className="rounded-xl border p-4 mb-8 bg-card">
      <h3 className="font-semibold mb-2">Quick Actions</h3>
      <div className="flex gap-2 flex-wrap">
        <button className="bg-primary text-white rounded px-4 py-2">Invite User</button>
        <button className="bg-secondary text-black rounded px-4 py-2">Add Team</button>
        <button className="bg-muted text-black rounded px-4 py-2">Change Logo</button>
      </div>
    </div>
  );
} 