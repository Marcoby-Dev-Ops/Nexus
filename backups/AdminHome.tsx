import React from 'react';
import { Users, Settings, Shield, Activity, Database, Server, AlertTriangle, CheckCircle } from 'lucide-react';
import { PageTemplates } from '@/components/patterns/PageTemplates';
import { ContentCard } from '@/components/patterns/ContentCard';
import { KpiCard } from '@/components/dashboard/KpiCard';
import { SimpleBarChart } from '@/components/dashboard/SimpleBarChart';

/**
 * @name AdminHome
 * @description Admin panel dashboard for managing NEXUS settings and system administration.
 * @returns {JSX.Element} The rendered AdminHome component.
 */

// Sample data for the admin dashboard
const adminKpiData = [
  { title: 'Active Users', value: 24, delta: '+3', trend: 'up' },
  { title: 'System Uptime', value: '99.9%', delta: '+0.1%', trend: 'up' },
  { title: 'Storage Used', value: '1.2TB', delta: '+8.5%', trend: 'up' },
  { title: 'API Calls', value: '45.2K', delta: '+12.3%', trend: 'up' },
];

const userActivityData = [
  { name: 'Mon', value: 18 },
  { name: 'Tue', value: 22 },
  { name: 'Wed', value: 19 },
  { name: 'Thu', value: 24 },
  { name: 'Fri', value: 21 },
  { name: 'Sat', value: 8 },
  { name: 'Sun', value: 6 },
];

const systemHealthData = [
  { name: 'CPU', value: 65 },
  { name: 'Memory', value: 78 },
  { name: 'Storage', value: 45 },
  { name: 'Network', value: 32 },
];

const recentUsers = [
  { name: 'John Doe', email: 'john@company.com', role: 'Admin', lastActive: '2 minutes ago', status: 'online' },
  { name: 'Jane Smith', email: 'jane@company.com', role: 'Manager', lastActive: '15 minutes ago', status: 'online' },
  { name: 'Mike Johnson', email: 'mike@company.com', role: 'User', lastActive: '1 hour ago', status: 'offline' },
  { name: 'Sarah Wilson', email: 'sarah@company.com', role: 'User', lastActive: '2 hours ago', status: 'offline' },
];

const systemAlerts = [
  { type: 'warning', message: 'High memory usage detected on server 2', time: '5 minutes ago' },
  { type: 'info', message: 'Scheduled backup completed successfully', time: '1 hour ago' },
  { type: 'success', message: 'System update installed successfully', time: '2 hours ago' },
  { type: 'warning', message: 'SSL certificate expires in 30 days', time: '1 day ago' },
];

const quickActions = [
  { label: 'User Management', icon: <Users className="w-5 h-5" />, onClick: () => console.log('User Management') },
  { label: 'System Settings', icon: <Settings className="w-5 h-5" />, onClick: () => console.log('System Settings') },
  { label: 'Security Center', icon: <Shield className="w-5 h-5" />, onClick: () => console.log('Security Center') },
  { label: 'System Logs', icon: <Activity className="w-5 h-5" />, onClick: () => console.log('System Logs') },
];

const AdminHome: React.FC = () => {
  return (
    <PageTemplates.Department
      title="Admin Panel"
      subtitle="Manage your NEXUS settings, users, and system administration"
    >
      {/* Quick Actions */}
      <ContentCard title="Quick Actions" variant="elevated" className="mb-8">
        <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
          {quickActions.map((action, index) => (
            <button
              key={index}
              onClick={action.onClick}
              className="flex flex-col items-center p-4 rounded-lg border border-border hover:border-primary hover:bg-primary/5 transition-all duration-200 group"
            >
              <div className="p-4 rounded-lg bg-primary/10 text-primary group-hover:bg-primary group-hover:text-primary-foreground transition-all duration-200 mb-3">
                {action.icon}
              </div>
              <span className="text-sm font-medium text-foreground group-hover:text-primary transition-colors duration-200">
                {action.label}
              </span>
            </button>
          ))}
        </div>
      </ContentCard>

      {/* KPI Cards */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6 mb-8">
        {adminKpiData.map((kpi, index) => (
          <KpiCard key={index} {...kpi} />
        ))}
      </div>

      {/* System Health and User Activity */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 mb-8">
        <ContentCard 
          title="User Activity" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Daily active users (last 7 days)</p>
          </div>
          <SimpleBarChart data={userActivityData} />
        </ContentCard>

        <ContentCard 
          title="System Health" 
          variant="elevated"
          className="hover:shadow-xl transition-all duration-300"
        >
          <div className="mb-6">
            <p className="text-sm text-muted-foreground">Resource utilization percentage</p>
          </div>
          <SimpleBarChart data={systemHealthData} />
        </ContentCard>
      </div>

      {/* Recent Users and System Alerts */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-8">
        <ContentCard 
          title="Recent Users" 
          variant="elevated"
          action={
            <button className="px-4 py-2 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              Manage Users
            </button>
          }
        >
          <div className="space-y-4">
            {recentUsers.map((user, index) => (
              <div key={index} className="flex items-center justify-between p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                <div className="flex items-center space-x-4">
                  <div className="p-2 rounded-lg bg-primary/10">
                    <Users className="w-5 h-5 text-primary" />
                  </div>
                  <div>
                    <h4 className="font-medium text-foreground">{user.name}</h4>
                    <p className="text-sm text-muted-foreground">{user.email} â€¢ {user.role}</p>
                  </div>
                </div>
                <div className="text-right">
                  <div className={`inline-flex px-2 py-1 rounded-full text-xs font-medium ${
                    user.status === 'online' ? 'bg-success/10 text-success' : 'bg-muted text-foreground'
                  }`}>
                    {user.status}
                  </div>
                  <p className="text-xs text-muted-foreground mt-1">{user.lastActive}</p>
                </div>
              </div>
            ))}
          </div>
        </ContentCard>

        <ContentCard 
          title="System Alerts" 
          variant="elevated"
          action={
            <button className="px-4 py-2 bg-primary/5 text-primary hover:bg-primary/10 hover:text-primary/90 rounded-lg transition-all duration-200 text-sm font-medium border border-border">
              View All Alerts
            </button>
          }
        >
          <div className="space-y-4">
            {systemAlerts.map((alert, index) => {
              const getIcon = (type: string) => {
                switch (type) {
                  case 'warning': return <AlertTriangle className="w-5 h-5" />;
                  case 'success': return <CheckCircle className="w-5 h-5" />;
                  case 'info': return <Activity className="w-5 h-5" />;
                  default: return <Activity className="w-5 h-5" />;
                }
              };

              const getColor = (type: string) => {
                switch (type) {
                  case 'warning': return 'bg-warning/10 text-warning';
                  case 'success': return 'bg-success/10 text-success';
                  case 'info': return 'bg-primary/10 text-primary';
                  default: return 'bg-muted text-muted-foreground';
                }
              };

              return (
                <div key={index} className="flex items-start space-x-4 p-4 rounded-lg hover:bg-muted/50 transition-colors duration-200">
                  <div className={`p-2 rounded-lg ${getColor(alert.type)}`}>
                    {getIcon(alert.type)}
                  </div>
                  <div className="flex-1">
                    <p className="text-sm text-foreground">{alert.message}</p>
                    <p className="text-xs text-muted-foreground mt-1">{alert.time}</p>
                  </div>
                </div>
              );
            })}
          </div>
        </ContentCard>
      </div>
    </PageTemplates.Department>
  );
};

export default AdminHome; 