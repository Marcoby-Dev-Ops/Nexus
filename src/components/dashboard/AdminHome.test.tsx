/**
 * @file AdminHome.test.tsx
 * @description Unit and snapshot tests for the AdminHome component.
 */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AdminHome from './AdminHome';

describe('AdminHome', () => {
  it('renders all major sections', () => {
    render(<AdminHome />);
    
    // Check for section headers
    expect(screen.getByText('Quick Actions')).toBeInTheDocument();
    expect(screen.getByText('User Management')).toBeInTheDocument();
    expect(screen.getByText('Team Management')).toBeInTheDocument();
    expect(screen.getByText('System Settings')).toBeInTheDocument();
    expect(screen.getByText('Activity Log')).toBeInTheDocument();
    expect(screen.getByText('AI Insights')).toBeInTheDocument();
  });

  it('renders user management table with correct data', () => {
    render(<AdminHome />);
    
    // Check table headers
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('Email')).toBeInTheDocument();
    expect(screen.getByText('Role')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();

    // Check user data
    expect(screen.getByText('Alice')).toBeInTheDocument();
    expect(screen.getByText('alice@email.com')).toBeInTheDocument();
    expect(screen.getByText('Admin')).toBeInTheDocument();
    expect(screen.getByText('Active')).toBeInTheDocument();
  });

  it('renders team management list with correct data', () => {
    render(<AdminHome />);
    
    // Check team data
    expect(screen.getByText('Sales')).toBeInTheDocument();
    expect(screen.getByText('5 members')).toBeInTheDocument();
    expect(screen.getByText('Finance')).toBeInTheDocument();
    expect(screen.getByText('3 members')).toBeInTheDocument();
  });

  it('renders system settings with correct data', () => {
    render(<AdminHome />);
    
    // Check system settings
    expect(screen.getByText('Branding:')).toBeInTheDocument();
    expect(screen.getByText('Nexus OS')).toBeInTheDocument();
    expect(screen.getByText('Integrations:')).toBeInTheDocument();
    expect(screen.getByText('QuickBooks, Slack')).toBeInTheDocument();
    expect(screen.getByText('Preferences:')).toBeInTheDocument();
    expect(screen.getByText('Dark mode enabled')).toBeInTheDocument();
  });

  it('renders activity log with correct data', () => {
    render(<AdminHome />);
    
    // Check activity log entries
    expect(screen.getByText('Invited user')).toBeInTheDocument();
    expect(screen.getByText('Alice')).toBeInTheDocument();
    expect(screen.getByText('(2024-06-10)')).toBeInTheDocument();
    expect(screen.getByText('Changed logo')).toBeInTheDocument();
    expect(screen.getByText('Bob')).toBeInTheDocument();
    expect(screen.getByText('(2024-06-09)')).toBeInTheDocument();
  });

  it('renders AI insights with correct data', () => {
    render(<AdminHome />);
    
    // Check AI insights
    expect(screen.getByText('Carol has not accepted her invite (3 days).')).toBeInTheDocument();
    expect(screen.getByText('No security issues detected.')).toBeInTheDocument();
  });

  it('renders quick action buttons', () => {
    render(<AdminHome />);
    
    // Check quick action buttons
    expect(screen.getByText('Invite User')).toBeInTheDocument();
    expect(screen.getByText('Add Team')).toBeInTheDocument();
    expect(screen.getByText('Change Logo')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<AdminHome />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 