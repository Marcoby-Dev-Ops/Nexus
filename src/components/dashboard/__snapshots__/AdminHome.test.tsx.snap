// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHome matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-8 space-y-8"
  >
    <div
      class="rounded-xl border p-4 mb-8 bg-card"
    >
      <h3
        class="font-semibold mb-2"
      >
        Quick Actions
      </h3>
      <div
        class="flex gap-2 flex-wrap"
      >
        <button
          class="bg-primary text-white rounded px-4 py-2"
        >
          Invite User
        </button>
        <button
          class="bg-secondary text-black rounded px-4 py-2"
        >
          Add Team
        </button>
        <button
          class="bg-muted text-black rounded px-4 py-2"
        >
          Change Logo
        </button>
      </div>
    </div>
    <div
      class="rounded-xl border p-4 bg-card"
    >
      <h3
        class="font-semibold mb-2"
      >
        User Management
      </h3>
      <table
        class="min-w-full text-sm"
      >
        <thead>
          <tr>
            <th
              class="text-left p-2"
            >
              Name
            </th>
            <th
              class="text-left p-2"
            >
              Email
            </th>
            <th
              class="text-left p-2"
            >
              Role
            </th>
            <th
              class="text-left p-2"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Alice
            </td>
            <td
              class="p-2"
            >
              alice@email.com
            </td>
            <td
              class="p-2"
            >
              Admin
            </td>
            <td
              class="p-2"
            >
              Active
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Bob
            </td>
            <td
              class="p-2"
            >
              bob@email.com
            </td>
            <td
              class="p-2"
            >
              Manager
            </td>
            <td
              class="p-2"
            >
              Active
            </td>
          </tr>
          <tr
            class="border-t"
          >
            <td
              class="p-2"
            >
              Carol
            </td>
            <td
              class="p-2"
            >
              carol@email.com
            </td>
            <td
              class="p-2"
            >
              User
            </td>
            <td
              class="p-2"
            >
              Invited
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="rounded-xl border p-4 bg-card"
    >
      <h3
        class="font-semibold mb-2"
      >
        Team Management
      </h3>
      <ul
        class="divide-y"
      >
        <li
          class="py-2 flex justify-between"
        >
          <span>
            Sales
          </span>
          <span
            class="text-xs text-muted-foreground dark:text-gray-200"
          >
            5 members
          </span>
        </li>
        <li
          class="py-2 flex justify-between"
        >
          <span>
            Finance
          </span>
          <span
            class="text-xs text-muted-foreground dark:text-gray-200"
          >
            3 members
          </span>
        </li>
        <li
          class="py-2 flex justify-between"
        >
          <span>
            Operations
          </span>
          <span
            class="text-xs text-muted-foreground dark:text-gray-200"
          >
            4 members
          </span>
        </li>
      </ul>
    </div>
    <div
      class="rounded-xl border p-4 bg-card"
    >
      <h3
        class="font-semibold mb-2"
      >
        System Settings
      </h3>
      <ul
        class="list-disc pl-5"
      >
        <li>
          Branding: 
          <span
            class="text-muted-foreground dark:text-gray-200"
          >
            Nexus OS
          </span>
        </li>
        <li>
          Integrations: 
          <span
            class="text-muted-foreground dark:text-gray-200"
          >
            QuickBooks, Slack
          </span>
        </li>
        <li>
          Preferences: 
          <span
            class="text-muted-foreground dark:text-gray-200"
          >
            Dark mode enabled
          </span>
        </li>
      </ul>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div
        class="rounded-xl border p-4 bg-card"
      >
        <h3
          class="font-semibold mb-2"
        >
          Activity Log
        </h3>
        <ul
          class="divide-y"
        >
          <li
            class="py-2"
          >
            Invited user by 
            <span
              class="font-medium"
            >
              Alice
            </span>
             
            <span
              class="text-xs text-muted-foreground dark:text-gray-200"
            >
              (2024-06-10)
            </span>
          </li>
          <li
            class="py-2"
          >
            Changed logo by 
            <span
              class="font-medium"
            >
              Bob
            </span>
             
            <span
              class="text-xs text-muted-foreground dark:text-gray-200"
            >
              (2024-06-09)
            </span>
          </li>
        </ul>
      </div>
      <div
        class="rounded-xl border p-4 bg-card"
      >
        <h3
          class="font-semibold mb-2"
        >
          AI Insights
        </h3>
        <ul
          class="list-disc pl-5"
        >
          <li>
            Carol has not accepted her invite (3 days).
          </li>
          <li>
            No security issues detected.
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
